src/amulware.Graphics/Core/vertexdata/IVertexData.cs

```diff
@@ -11,11 +11,5 @@ public interface IVertexData
         /// </summary>
         /// <returns>Array of <see cref="VertexAttribute"/></returns>
         VertexAttribute[] VertexAttributes();
```

Comment:
While backwards-incompatible changes are introduced: is there a reason this is not a getter?

Reply:
I had not thought about that.

Thinking about it now, I'm not sure if it should be. While you could implement it differently, we always implement it as creating the attribute array when being called, which is a relatively large amount of work (it creates a list of attribute templates, then turns them all into attributes with correct offset and step and then returns those in an array). I'm not sure if a property should do that much work.

This method is also not called a lot, i.e. the return value of it is not used over and over, but only once per vertex buffer to set up the vertex array. A property tells me that it's a value that is expected to be read a lot.

What do you think?

Reply:
That makes sense, though I would then expect the method to be called `MakeVertexAttributes` or `CreateVertexAttributes`, since it really takes on the signature of a factory method in that case (note that some languages opt to remove these filler words such as "make" and "create", but I believe C# is of the more obtuse variant of programming languages).

I am not sure if such a minor change warrants a backwards incompatibility change though.

Reply:
I agree that using a verb in the name would be better, yes. I'll think about it a bit more. In the end, I'm still not 100% happy with this interface, since this method doesn't actually return any information about the vertex instance, but about the vertex type.